dotnet/notebooks/06-vector-stores-and-embeddings.ipynb

```diff
@@ -0,0 +1,464 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Upsert records\n",
+    "\n",
+    "Now our glossary records are ready to be inserted into database. For that, we can use `collection.UpsertAsync` or `collection.UpsertBatchAsync` methods. Note that this operation is idempotent - if record with specific key doesn't exist, it will be inserted. If it already exists, it will be updated. As a result, we should receive a keys of recently upserted records:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "await foreach (var key in collection.UpsertBatchAsync(glossaryEntries))\n",
+    "{\n",
+    "    Console.WriteLine(key);\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Get records by key\n",
+    "\n",
+    "In order to ensure our records were upserted correctly, we can get these records by a key with `collection.GetAsync` or `collection.GetBatchAsync` methods. \n",
+    "\n",
+    "Both methods accept `GetRecordOptions` class as a parameter, where you can specify if you want to include vector properties in your response or not. Taking into account that vector dimension value can be high, if you don't need to work with vectors in your code, it's recommended to not fetch them from the database. That's why `GetRecordOptions.IncludeVectors` property is `false` by default. \n",
+    "\n",
+    "In this example, we want to include vectors in a result to ensure that our data was upserted correctly:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "var options = new GetRecordOptions() { IncludeVectors = true };\n",
+    "\n",
+    "await foreach (var record in collection.GetBatchAsync(keys: [1, 2, 3], options))\n",
+    "{\n",
+    "    Console.WriteLine($\"Key: {record.Key}\");\n",
+    "    Console.WriteLine($\"Term: {record.Term}\");\n",
+    "    Console.WriteLine($\"Definition: {record.Definition}\");\n",
+    "    Console.WriteLine($\"Definition Embedding: {JsonSerializer.Serialize(record.DefinitionEmbedding)}\");\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Perform a search\n",
+    "\n",
+    "Since we ensured that our records are already in a database, we can perform a vector search with `collection.VectorizedSearchAsync` method. \n",
+    "\n",
+    "This method accepts `VectorSearchOptions` class as a parameter, which allows to configure your vector search operation - specify the maximum number of records to return, the number of results to skip before returning results, a search filter to use before doing the vector search and so on. More information about it can be found [here](https://learn.microsoft.com/en-us/semantic-kernel/concepts/vector-store-connectors/vector-search#vector-search-options).\n",
```

Comment:
```suggestion
    "This method accepts the `VectorSearchOptions` class as a parameter, which allows configuration of the vector search operation - specify the maximum number of records to return, the number of results to skip before returning results, a search filter to use before doing the vector search and so on. More information about it can be found [here](https://learn.microsoft.com/en-us/semantic-kernel/concepts/vector-store-connectors/vector-search#vector-search-options).\n",
```